rotkehlchen/chain/ethereum/modules/yearn/v2_vaults.py

```diff
@@ -0,0 +1,237 @@
+                vault_token_symbol = event.to_asset.identifier
+                underlying_token = event.from_asset
+            else:
+                vault_token_symbol = event.from_asset.identifier
+                underlying_token = event.to_asset
+
+            vaults[vault_token_symbol]['events'].append(event)
+
+        # Sort events in each vault
+        for key in vaults.keys():
+            vaults[key]['events'].sort(key=lambda x: x.timestamp)
+            total_pnl = self._process_vault_events(vaults[key]['events'])
+
+            current_balance = None
+            for balance in defi_balances:
+                found_balance = (
+                    balance.protocol.name == 'yearn.finance • Vaults' and
+                    balance.base_balance.token_symbol == vault_token_symbol
+                )
+                if found_balance:
+                    current_balance = balance.underlying_balances[0].balance
+                    total_pnl += current_balance
+                    break
+
+                # Due to the way we calculate usd prices for vaults we need to get the current
+                # usd price of the actual pnl amount at this point
+                if total_pnl.amount != ZERO:
+                    usd_price = get_usd_price_zero_if_error(
+                        asset=underlying_token,
+                        time=ts_now(),
+                        location='yearn vault history',
+                        msg_aggregator=self.msg_aggregator,
+                    )
+                    total_pnl.usd_value = usd_price * total_pnl.amount
+
+            vaults_histories[key] = YearnVaultHistory(
+                events=vaults[key]['events'],
+                profit_loss=total_pnl,
+            )
+
+        return vaults_histories
+
+    def get_history(
+            self,
+            given_defi_balances: 'GIVEN_DEFI_BALANCES',
+            addresses: List[ChecksumEthAddress],
+            reset_db_data: bool,
+            from_timestamp: Timestamp,  # pylint: disable=unused-argument
+            to_timestamp: Timestamp,  # pylint: disable=unused-argument
+    ) -> Dict[ChecksumEthAddress, Dict[str, YearnVaultHistory]]:
+        with self.history_lock:
+
+            # TODO: Skip entries in DB already queried. Since is a graph query
+            # it doesn't have a huge performance effect but is less data we have
+            # to process
+
+            if reset_db_data is True:
+                self.database.delete_yearn_vaults_data()
+
+            if isinstance(given_defi_balances, dict):
+                defi_balances = given_defi_balances
+            else:
+                defi_balances = given_defi_balances()
+
+            from_block = self.ethereum.get_blocknumber_by_time(from_timestamp)
+            to_block = self.ethereum.get_blocknumber_by_time(to_timestamp)
+            history: Dict[ChecksumEthAddress, Dict[str, YearnVaultHistory]] = {}
+
+            for address in addresses:
```

Comment:
I wanted to translate the same behavior from v1 code but since we are querying the graph we can retrieve all the information in just one call. Fix this